plugins/user-authenticators/ldap/src/org/apache/cloudstack/api/command/LdapCreateAccountCmd.java

```diff
@@ -119,6 +131,9 @@ private Long getDomainId() {
 
     @Override
     public void execute() throws ServerApiException {
+        if (getAccountType() == null && getRoleId() == null) {
+            throw new ServerApiException(ApiErrorCode.PARAM_ERROR, "Both account type and role ID are not provided");
```

Comment:
@jburwell before this changes `accountType` was required, now it's made optional with introduction of roleId, this check is to ensure at least one of the two are provided.